src/rules/interpolation-data.js

```diff
@@ -0,0 +1,74 @@
+const minimatch = require('minimatch');
+const { getKeyValue, get, getLangConfig } = require('../utils/utils');
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'ensures that used translate key is in translation file',
+      category: 'Possible errors',
+    },
+    schema: [
+      {
+        type: 'object',
+        required: ['interpolationPattern'],
+        properties: {
+          interpolationPattern: {
+            type: 'string',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  create(context) {
+    const config = Object.assign({}, context.settings.i18n, context.options[0]);
+
+    if (!config || (config.ignoreFiles && minimatch(context.getFilename(), config.ignoreFiles))) {
+      return {};
+    }
+
+    return {
+      CallExpression(node) {
+        const funcName = (node.callee.type === 'MemberExpression' && node.callee.property.name) || node.callee.name;
+
+        if (funcName !== config.functionName || !node.arguments || !node.arguments.length) {
+          return;
+        }
+
+        const [keyNode, dataNode] = node.arguments;
+        const key = getKeyValue(keyNode);
+
+        if (!key) {
+          return;
+        }
+
+        getLangConfig(config, 'principalLangs').forEach(({ translation }) => {
+          if (!translation) {
+            return;
+          }
+
+          const value = get(translation, key);
+          const interpolationTester = new RegExp(config.interpolationPattern);
+
+          if (value && (!dataNode || dataNode.name === 'undefined') && interpolationTester.test(value)) {
```

Comment:
It would be quite hard, it could only work if data parameter is a plain object and not a variable identifier and it doesn't seem like the common case. I could add an additional check for this though